syntax = "proto3";

package NProto;

message TPutRequest {
    uint64 request_id = 1;
    string key = 2;
    uint64 offset = 3;
}

message TPutResponse {
    uint64 request_id = 1;
}

message TGetRequest {
    uint64 request_id = 1;
    string key = 2;
}

message TGetResponse {
    uint64 request_id = 1;
    uint64 offset = 2;
}

message TInsertRequest {
    uint64 request_id = 1;
    string key = 2;
    string val = 3;
}

message TInsertResponse {
    uint64 request_id = 1;
}

message TGetInsertedRequest {
    uint64 request_id = 1;
    string key = 2;
}

message TGetInsertedResponse {
    uint64 request_id = 1;
    string value = 2;
}